qek/data/extractors.py

```diff
@@ -351,65 +351,77 @@ def __init__(
 
         super().__init__(path=path, device=device, compiler=compiler)
         self._sdk = sdk
-        self._batch_id = batch_id
+        self._batch_ids: list[str] | None = batch_ids
 
     @property
     def batch_ids(self) -> list[str] | None:
-        return self._batch_id
+        return self._batch_ids
 
     async def run(self) -> list[ProcessedData]:
         if len(self.sequences) == 0:
             logger.warning("No sequences to run, did you forget to call compile()?")
             return []
 
-        batches: list[Batch] = []
-        if self._batch_id is None:
+        if self._batch_ids is None:
             # Enqueue jobs.
             self._batch_ids = []
             for compiled in self.sequences:
-                logger.debug("Executing compiled graph #%s", id)
+                logger.debug("Enqueuing execution of compiled graph #%s", compiled.graph.id)
                 batch = self._sdk.create_batch(
                     compiled.sequence.to_abstract_repr(),
-                    jobs=[{"runs": 1000}],
+                    # Note: The SDK API doesn't support runs longer than 500 jobs.
```

Comment:
nit: this is not a limitation from the SDK, it comes from the specs of Fresnel itself

Reply:
Well, the SDK could automatically divide longer runs.

Do you know if this constant is exposed somewhere?

Reply:
It is stored in the device object under the `max_runs` attribute